tests/dyn_table_name_tests.rs

```diff
@@ -0,0 +1,79 @@
+pub mod common;
+
+pub use common::{features::*, setup::*, TestContext};
+use pretty_assertions::assert_eq;
+use sea_orm::{
+    entity::prelude::*, DatabaseConnection, Delete, IntoActiveModel, Iterable, QueryTrait, Set,
+    Update,
+};
+use sea_query::{Expr, IntoIden, Query};
+
+#[sea_orm_macros::test]
+#[cfg(any(
+    feature = "sqlx-mysql",
+    feature = "sqlx-sqlite",
+    feature = "sqlx-postgres"
+))]
+async fn main() -> Result<(), DbErr> {
+    let ctx = TestContext::new("dyn_table_name_tests").await;
+    create_tables(&ctx.db).await?;
+    dyn_table_name_lazy_static(&ctx.db).await?;
+    ctx.delete().await;
+
+    Ok(())
+}
+
+pub async fn dyn_table_name_lazy_static(db: &DatabaseConnection) -> Result<(), DbErr> {
+    use dyn_table_name_lazy_static::*;
+
+    for i in 1..=2 {
+        let entity = Entity {
+            table_name: TableName::from_str_truncate(format!("dyn_table_name_lazy_static_{}", i)),
+        };
+
+        let model = Model {
+            id: 1,
+            name: "1st Row".into(),
+        };
+        // Prepare insert statement
+        let mut insert = Entity::insert(model.clone().into_active_model());
+        // Reset the table name of insert statement
+        insert.query().into_table(entity.table_ref());
+        // Execute the insert statement
+        assert_eq!(insert.exec(db).await?.last_insert_id, 1);
+
+        // Prepare select statement
+        let mut select = Entity::find();
```

Comment:
Is there any way to use the Entity directly in full ORM style, like `Entity::find_by_id(0).one(&db).await?;` here? Or is that lost when we have to initialize the Entity with data?

Reply:
> Or is that lost when we have to initialize the Entity with data?

This is correct. Unfortunately the table name data is lost and we need to set it manually.